### PR TITLE
Bump to the most recent ledger

### PR DESCRIPTION
**Pull request by nc6**
This updates to the most recent ledger version.

- Removes some JSON instances now defined in ledger.
- Deals with the move of many protocol related concerns to the `cardano-protocol-tpraos` package.
- Updates with changes in ledger state (use of `SplitMap` and `UMap`).

Various of the changes here are not optimal (e.g. the proliferation of `DSignable` constraints). This is because this code will be soon revised anyway, as part of splitting the protocol fully away from the ledger inside the Shelley codebase. This work is largely extracted from the open work on that.

### CODE REVIEW

File: ouroboros-consensus-cardano/tools/db-analyser/Block/Alonzo.hs [hunk 1]
```diff
@@ -1,39 +1,25 @@
-{-# LANGUAGE DeriveAnyClass             #-}
-{-# LANGUAGE DerivingStrategies         #-}
-{-# LANGUAGE FlexibleInstances          #-}
-{-# LANGUAGE GeneralizedNewtypeDeriving #-}
-{-# LANGUAGE NamedFieldPuns             #-}
-{-# LANGUAGE OverloadedStrings          #-}
-{-# LANGUAGE StandaloneDeriving         #-}
-{-# LANGUAGE TypeApplications           #-}
-{-# LANGUAGE TypeFamilies               #-}
+
```
Comment (nfrisby) [inline ouroboros-consensus-cardano/tools/db-analyser/Block/Alonzo.hs:1]: Please remove this empty line 1.

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Mempool.hs [hunk 2]
```diff
@@ -283,22 +284,22 @@ theLedgerLens f x =
...
 -------------------------------------------------------------------------------}
 
-instance (SL.PraosCrypto c) => TxLimits (ShelleyBlock (ShelleyEra c)) where
+instance (ShelleyBasedEra (MaryEra c)) => TxLimits (ShelleyBlock (ShelleyEra c)) where
   type TxMeasure (ShelleyBlock (ShelleyEra c)) = ByteSize
   txMeasure        = ByteSize . txInBlockSize . txForgetValidated
   txsBlockCapacity = ByteSize . txsMaxBytes
 
-instance (SL.PraosCrypto c) => TxLimits (ShelleyBlock (AllegraEra c)) where
+instance (ShelleyBasedEra (MaryEra c)) => TxLimits (ShelleyBlock (AllegraEra c)) where
```
Comment (nfrisby) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Mempool.hs:292]: Copy-and paste error, `MaryEra`?

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Mempool.hs [hunk 3]
```diff
@@ -283,22 +284,22 @@ theLedgerLens f x =
   Tx Limits
 -------------------------------------------------------------------------------}
 
-instance (SL.PraosCrypto c) => TxLimits (ShelleyBlock (ShelleyEra c)) where
+instance (ShelleyBasedEra (MaryEra c)) => TxLimits (ShelleyBlock (ShelleyEra c)) where
```
Comment (nfrisby) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Mempool.hs:287]: Copy-and paste error, `MaryEra`?

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/TPraos.hs [hunk 4]
```diff
@@ -1,8 +1,10 @@
...
-{-# LANGUAGE TypeFamilies        #-}
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE FlexibleInstances    #-}
+{-# LANGUAGE ScopedTypeVariables  #-}
+{-# LANGUAGE TypeFamilies         #-}
 {-# OPTIONS_GHC -Wno-orphans #-}
+{-# LANGUAGE FlexibleContexts     #-}
+{-# LANGUAGE UndecidableInstances #-}
 
 -- TODO where to put this?
```
Comment (nfrisby) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/TPraos.hs:9]: Maybe we can remove this TODO now? 😁

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs [hunk 5]
```diff
@@ -505,9 +510,8 @@ registerGenesisStaking staking nes = nes {
     -- establish an initial stake distribution.
     initSnapShot :: SL.SnapShot (EraCrypto era)
     initSnapShot =
-        SL.stakeDistr
-          @era
-          (SL._utxo (SL._utxoState ledgerState))
+        SL.incrementalStakeDistr
```
Comment (nfrisby) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs:513]: It's not obvious to me that this is correct. Maybe some comments re the `mempty` parts and the new use of the term "incremental"?

**Review by nfrisby — APPROVED** [review]
As the PR description says, most everything is adjusting only imports or adding the `DSign` constraint throughout. Which is fine, as it's a temporary symptom.

I caught what looks like a few copy-paste errors and asked a couple questions. Nothing major.

**Review by nfrisby — CHANGES_REQUESTED** [review]
Whoops, I meant Request Changes, not Approve! Just for the suspected copy-paste errors.

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Mempool.hs (hunk 2 above)
Comment (nc6) [reply ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Mempool.hs:292]: Whoops, not sure how I managed that! Good catch

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs (hunk 5 above)
Comment (nfrisby) [reply ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs:518]: The comments are good 👍 , but a git merge conflict was saved into the file.

**Comment by nc6** [comment]
bors merge

**Pull request merged**